commodore/dependency_mgmt.py

```diff
@@ -96,21 +96,13 @@ def _read_component_urls(cfg: Config, component_names) -> Dict[str, str]:
             f"Could not read Commodore configuration: {e}"
         ) from e
 
-    for component_name in component_names:
-        component_urls[
-            component_name
-        ] = f"{cfg.default_component_base}/{component_name}.git"
-
-    if commodore_config is None:
-        click.secho("Empty Commodore config file", fg="yellow")
-    else:
-        for component_override in commodore_config.get("components", []):
-            if cfg.debug:
-                click.echo(
-                    f"   > Found override for component {component_override['name']}:"
-                )
-                click.echo(f"     Using URL {component_override['url']}")
-            component_urls[component_override["name"]] = component_override["url"]
+    for component in commodore_config.get("components", []):
+        if component["name"] in component_names:
+            component_urls[component["name"]] = component["url"]
+
+    for name in component_names:
+        if name not in component_urls:
+            raise click.ClickException(f"No url for component '{name}' configured")
```

Comment:
It would be nice if we could allow to not specify a component in `commodore.yml` but instead use `component_versions` in the hierarchy to set the URL. This would allow to add new components for only a part of the hierarchy (i.e. tenant/cluster specific components).

I think it would be enough to just move this check to after discovering component version overrides.

Reply:
Ok, it's not quite that easy. Do you want to do this in this PR or create a new one/issue?

Reply:
It would require iterating `cluster_inventory["parameters"].get("component_versions", None)` to get all URLs from there as well, before cloning the components.

Reply:
I consider this outside of the original change scope. Can you please create a follow up issue for this?